### ISSUE TITLE
Please support multiple frames in PGM file format

### ISSUE THREAD

**Issue by BarchSteel** [imagej/imagej#268]
The PGM file format (http://netpbm.sourceforge.net/doc/pgm.html) allows for multiple frames in one file, called a sequence.
The frames are just written, without gaps or padding, sequentially to the file.
The number of frames in the file can be determined by the (fileSize - Header)/(size_of_one_frame)

If I open such a file in ImageJ, only the first frame is displayed.
Please improve the PGM file reader, so it can read multiple frames as a sequence, just like the RAW importer does.
[sequence_3_frames_38x38px_8bit.zip](https://github.com/imagej/imagej/files/5675211/sequence_3_frames_38x38px_8bit.zip)

As a work-around, you can import the file using the "RAW" importer. For example, for the attached file, use uint8, size: 38x38 px, 3 frames.

**Comment by imagejan** [comment]
That's an issue for the `ome/bioformats` repository, as [`PGMReader`](https://github.com/ome/bioformats/blob/0caf95dca09609938875b59d805422652090faa8/components/formats-bsd/src/loci/formats/in/PGMReader.java) is implemented there.

**Comment by ctrueden** [comment]
To complicate matters: because PGM is an open format, there is also a SCIFIO implementation: [`PGMFormat`](https://github.com/scifio/scifio/blob/scifio-0.41.0/src/main/java/io/scif/formats/PGMFormat.java). Strategically, it is probably time for a conversation about whether to continue maintaining these SCIFIO implementations. Although there are many downsides to discontinuing it, resources-wise, having two implementations to keep synchronized is not really tenable at the moment.

**Comment by BarchSteel** [comment]
I opened bug report in SCIFIO: https://github.com/scifio/scifio/issues/445

**Comment by BarchSteel** [comment]
So, I played a little bit around with SCIFIO and patched the scifio PGMFormat.java to read multi-plane images.
On ImageJ 1.53c: (compiled from the GIT master branch from )
But this class is only called when I use File->Import->Image.
It would not load when I use File->Open Image.
So is / was there a 3rd class for PGM loading?

Now while writing this post, and testing it with ImageJ 1.53e, I can't find the menu option "File->Import->Image" any more. Its missing.

Now, I need to find out, where to get the sources for 1.53e and if it uses any different PGM loader.

**Comment by imagejan** [comment]
> It would not load when I use File->Open Image. ImageJ used some other class, and not my class.

Indeed, ImageJ uses Bio-Formats in that case, *unless* you specify "Use SCIFIO when opening files (BETA!)" in *Edit > Options > ImageJ2...*

> The git snapshot 2.1.1-SNAPSHOT shows in Help->"About ImageJ" the version 1.53c.

That's unfortunate, but a known issue: *Help > About ImageJ* is an *ImageJ1* command and displays the bundled *ImageJ1* version (i.e. the one of your `ij.jar` file).
Clicking on the info bar will tell you the versions of *both* IJ1 and IJ2, e.g. `(Fiji Is Just) ImageJ 2.1.0/1.53c`.

**Comment by BarchSteel** [comment]
> Indeed, ImageJ uses Bio-Formats in that case, _unless_ you specify "Use SCIFIO when opening files (BETA!)" in _Edit > Options > ImageJ2..._
That's an interesting comment. When I first read it, I though: well, when I compile ImageJ from source, there is no Bio-Format plug-in installed. But then you said that there is a bundled version of ImageJ1... and yes, in then dependency ij-1.5.3c.jar, there is another PGM reader at: ij.plugin.PGM_reader.class  (which does not seem to be part of the Bio-Formats package either).

Thank you, I will try your option "Use SCIFIO when opening files (BETA!)", it looks very promising ;-)
